projects/pastanaga/src/lib/translate/translate.pipe.ts

```diff
@@ -0,0 +1,33 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import * as translateKeys from '../../../../../src/assets/i18n/en.json';
+@Pipe({
+  name: 'translate'
+})
+export class TranslatePipe implements PipeTransform {
+    lastKey?: string;
+    lastParams?: string;
+    value: string = '';
+    transform(key: string, args?: any): any {
+        // if we ask another time for the same key, return the last value
+        if (key === this.lastKey && args === this.lastParams) {
+            return this.value;
+        }
+        const keys = !!key ? key.split('.') : [];
+        this.value = translateKeys['default'];
+        keys.map(k => {
```

Comment:
`map` is not meant to be used in this kind of case. `forEach` or `reduce` would make more sense.

By the way, you should never update `this.value` on each iteration like that. When you have a loop modify something, you need to modify a variable, and then assign the final result only to `this`. Otherwise you are triggering detectChange on every iteration of your loop.